### ISSUE TITLE
searchDisjunctiveFacetingAsync breaks on facet values with commas

### ISSUE THREAD

**Issue by PLNech** [algolia/algoliasearch-client-android#542]
- set `refinements={"lvl0":"Dom, byt a záhrada"}`
- set `disjunctiveFacets="lvl0"`
- call `searchDisjunctiveFacetingAsync(new Query().setFacets("*"), disjunctiveFacets, refinements, completionHandler);`

The query is sent with `facetFilters=(lvl0:Dom, byt a záhrada)`, which brings no result. Indeed, it should be `facetFilters=[\"lvl0:Dom, byt a záhrada\"]` which works as expected.

This prompts to a refactoring of searchDisjunctiveFacetingAsync, which was [already requested in a TODO](https://github.com/algolia/algoliasearch-client-android/blob/master/algoliasearch/src/main/java/com/algolia/search/saas/helpers/DisjunctiveFaceting.java#L118).

**Comment by PLNech** [comment]
@clement-leprovost, would love your take on this one before refactoring.

- Why the [parentheses](https://github.com/algolia/algoliasearch-client-android/blob/master/algoliasearch/src/main/java/com/algolia/search/saas/helpers/DisjunctiveFaceting.java#L125), considering they don't seem to be part of the [expected syntax for `facetFilters`](https://www.algolia.com/doc/api-reference/api-parameters/facetFilters/?language=javascript#apply-filters-on-a-search-query)? 
- Do you think we can/should migrate to `filters` as `facetFilters` is deprecated?

**Comment by PLNech** [comment]
In the meantime, workaround consists in computing manually `filters`:
```java
// Create the query, refinements, disjunctiveFacets
Query query = new Query().setFacets("*").setMaxValuesPerFacet(20);
java.util.HashMap<String, List<String>> refinements = new HashMap<>();
refinements.put("office_region", Collections.singletonList("Prešovský"));
refinements.put("lvl0", Collections.singletonList("Dom, byt a záhrada"));
Collection<String> disjunctiveFacets = Collections.singletonList("lvl0");

// Manually build the filters string using the refinements
StringJoiner joinAnd = new StringJoiner(" AND ");
for (Map.Entry<String, List<String>> entry : refinements.entrySet()) {
    StringJoiner joinOr = new StringJoiner(" OR ");
    String attribute = entry.getKey();
    for (String value : entry.getValue()) {
        joinOr.add(attribute + ":\"" + value + "\"");
    }
    joinAnd.add(joinOr.toString());
}
query.setFilters(joinAnd.toString());

// Trigger disjunctive search with empty refinements parameter
index.searchDisjunctiveFacetingAsync(query, disjunctiveFacets, new HashMap<String, ArrayList<String>>(), new CompletionHandler() {
    @Override
    public void requestCompleted(JSONObject jsonObject, AlgoliaException e) {
        if (jsonObject != null) {
            Log.e("searchDisjunctive", "requestCompleted success: " + jsonObject.optString("nbHits"));
        }
        if (e != null) Log.e("searchDisjunctive", "requestCompleted error: " + e);
    }
});
```

**Comment by clement-leprovost** [comment]
Indeed parentheses make no sense when used with `facetFilters`. The Git blame says it's me, but I just moved around some code. 😇

Using `filters` instead of `facetFilters` will bring other issues, as this parameter also handles numeric filters and tag filters, and its syntax is different. I would stick to `facetFilters` for the time being. It's not deprecated. We just [recommend](https://www.algolia.com/doc/api-reference/api-parameters/facetFilters/) using `filters` for manually input queries (because it supports a more user-friendly, SQL-like syntax); but `facetFilters` still makes a lot of sense for automatically generated filters.

**Comment by PLNech** [comment]
Fix released in [v3.21.13](https://github.com/algolia/algoliasearch-client-android/releases/tag/3.21.13).
